### PR TITLE
Refactor basic message checking code in consensus

### PR DESCRIPTION
**Pull request by rlan35**
## Test
2 shards, 110 consensus, 317 total TPS, 11 nodes 127.0.0.1, 55, 155.107

### CODE REVIEW

File: consensus/consensus.go [hunk 1]
```diff
@@ -195,6 +196,46 @@ func New(host p2p.Host, ShardID string, peers []p2p.Peer, leader p2p.Peer) *Cons
 	return &consensus
 }
 
+// Checks the basic meta of a consensus message.
+func (consensus *Consensus) checkConsensusMessage(message consensus_proto.Message, publicKey *bls.PublicKey) bool {
```
Comment (LeoHChen) [inline consensus/consensus.go]: please add test to those functions.

File: consensus/consensus.go [hunk 2]
```diff
@@ -195,6 +196,46 @@ func New(host p2p.Host, ShardID string, peers []p2p.Peer, leader p2p.Peer) *Cons
...
+
+	if !bytes.Equal(blockHash, consensus.blockHash[:]) {
+		utils.GetLogInstance().Warn("Wrong blockHash", "consensus", consensus)
+		return false
+	}
+	return true
+}
+
+// Gets the validator peer based on validator ID.
+func (consensus *Consensus) getValidatorPeerByID(validatorID uint32) *p2p.Peer {
```
Comment (LeoHChen) [inline consensus/consensus.go]: add test.

File: consensus/consensus.go [hunk 3]
```diff
@@ -229,6 +270,20 @@ func (consensus *Consensus) signMessage(message []byte) []byte {
 	return signature.Serialize()
 }
 
+// Sign on the consensus message signature field.
+func (consensus *Consensus) signConsensusMessage(message *consensus_proto.Message) error {
```
Comment (LeoHChen) [inline consensus/consensus.go]: add test.

Comment (rlan35) [reply consensus/consensus.go]: Added an issue.

**Pull request merged**